docs/spark.rst

```diff
@@ -0,0 +1,74 @@
+Running Spark on Toil
+=====================
+
```

Comment:
We need some info in the installation section on how you install docker, etc. to make this work. @hannes-ucsc may have opinions on this.